File: tests/e2e/cases/09-type-coverage/swap_bytes_u16/swap_bytes_u16_verified.cpp
/*
DEMO: 16-bit byte swap.

Returns `x` with its two bytes swapped. Exercises:
  * `uint16_t` parameter + return — width not seen by the existing
    integer suites.
  * Composition of left + right shifts at 16-bit width (lowers to
    `shl i16` / `lshr i16` followed by `or i16`).

Expected verdict: VERIFIED.
*/

#include <cstdint>

uint16_t swap_bytes_u16(uint16_t x) {
    return (uint16_t)((x >> 8) | (uint16_t)(x << 8));
}

File: tests/e2e/cases/09-type-coverage/swap_bytes_u16/swap_bytes_u16_verified.rs
// DEMO: 16-bit byte swap (Rust mirror of swap_bytes_u16_verified.cpp).
// Exercises u16 parameter + return, absent from earlier suites.

fn swap_bytes_u16(x: u16) -> u16 {
    (x >> 8) | (x << 8)
}
